File: ruby/coding_challenge.rb
require 'pry'

class AgeOrdering
  def names(arr)
    sorted = arr.sort { |x, y| x[1] <=> y[1] }
    sorted.map do |name|
      name[0]
    end
  end

  def name_and_ages(arr)
    sorted = arr.sort { |x, y| x[1] <=> y[1] }
    sorted.map do |person|
      "#{person[0]} (#{person[1]})"
    end
  end

  def hashify(arr)
    sorted = arr.sort { |x, y| x[1] <=> y[1] }
    sorted.each_with_object( {} ) do |person, result|
      (result[person[1]] ||= []) << person[0]
    end
  end
end

File: javascript/codingChallenge.js
class AgeOrdering {
  namify(arr) {
    const sorted = arr.sort( (a, b) => {
      if (a[1] < b[1]) return -1
      if (a[1] > b[1]) return 1;
      return 0
    })
    return sorted.map(person => {
      return person[0]
    })
  }
}

module.exports = AgeOrdering;
